### PR TITLE
Make MessageEditor an UnaryOperator instead of Consumer

### LINKED ISSUE #7: MessageEditor as UnaryOperator might be a better fit
Currently it is a Consumer and it can become challenging to provide a default implementation and allow to adjust it via inheritance, as Sunrise allows.

### PR DESCRIPTION
**Pull request by lauraluiz**
Closes #7

### CODE REVIEW

File: email-smtp/it/AsynchronousIntegrationTest.java [hunk 1]
```diff
@@ -59,10 +59,11 @@ public void sendTenMessagesAsynchronouslyWith1PooledThreadAsASmallPerformanceTes
         private boolean finished = false;
 
         @Override
-        public synchronized void edit(@Nonnull final MimeMessage message) throws Exception {
+        public synchronized MimeMessage edit(@Nonnull final MimeMessage message) throws Exception {
```
Comment (katmatt) [inline email-smtp/it/AsynchronousIntegrationTest.java:62]: Do you know why we use the `@Nonnull` annotation here? Normally the default is non null... But it's just a question and we shouldn't change it now 😉

**Review by katmatt — APPROVED** [review]
LGTM 👍 And how about adding yourself to the pom.xml developer section?

File: email-smtp/it/AsynchronousIntegrationTest.java (hunk 1 above)
Comment (lauraluiz) [reply email-smtp/it/AsynchronousIntegrationTest.java:62]: Developer's decision. As it's not sunrise framework itself, I guess it's fine not to have it following sunrise's guidelines if that's what they preferred :)

**Comment by lauraluiz** [comment]
Ah... maybe next time, with a more important change 😄

**Pull request merged**